### ISSUE TITLE
[data loss] METABUG: don't overwrite files

### ISSUE THREAD

**Issue by chrizilla** [yt-dlp/yt-dlp#6740]
## description of the problem

yt-dlp/youtube-dl is the only program(s) I know overwriting files without asking the user. Overwriting files is dangerous because it potentially causes **DATA LOSS**. Data loss is a very serious issue and should never occur!

All other programs I know warn the user if a file (image, video, document, etc.) already exists and ask the user if it should really be replaced (Y/N).

I hear you say that only "temporary files" are deleted:

_Originally posted by @pukkandan in https://github.com/yt-dlp/yt-dlp/issues/6705#issuecomment-1499870495_

>> files are unintentionally overwritten
>> This should **NEVER** occur with `--no-overwrites --no-post-overwrites` , no ?
>
> Temporary files are deleted if they can't be resumed, irrespective of any options.

I disagree with this notion. Downloads are not "temporary files" in the conventional sense. Other programs delete/overwrite (conventional) temporary files without user interaction only if they don't contain meaningful data for the user (i.e. scratch file, dump file, other files that ensure the programs operation, .... etc.). But the files yt-dlp overwrites are not of that sort but are downloads. So the data is meaningful to the user and potentially very valuable, especially if the source material is not available for download any longer and thus can't be obtained anymore.

For example, I downloaded the historic CNN livestream of President Trump's arrest and yt-dlp overwrote my download, which is now gone forever and unobtainable (#6705).

## related issues

I create this issue as a META bug subsuming all other instances where users have been affected by files overwritten by yt-dlp:

* #6501
* #6846 
* #6705 = livestream download should not overwrite previous downloads
* #6416
* #1467
* #2426 *"the audio and video files will overwrite each other"*
* #6228
* #5977
* #3160
* #2854

## how to solve the problem

I argue that files should never be overwritten without user consent!
Ask the user if files should get overwritten like all other programs do.

If for some reason you are not willing or unable to accept this common standard, then at least respect the user choice when then user explicitly tells yt-dlp `--no-overwrites --no-post-overwrites`.

The helpfile also is very clear: Do not overwrite **ANY** files!

```
--no-overwrites           Do not overwrite any files
--no-post-overwrites      Do not overwrite post-processed files
````

**Comment by pukkandan** [comment]
1. Explanation for each is given in the specific issue. Did you just search for "overwrite" and link all the issues found? Most of them have nothing to do with your description.
2. Pls do not skip issue template by using "convert to issue".
3. Replacing **temporary** files is an essential part of the program's function and cannot be changed or controlled by user.
4. The only part of the post I could agree with is not overwriting metadata files by default. However, that has been the default for many years and cannot be changed lightly. `--no-overwrites` covers it
> Ask the user if files should get overwritten like all other programs do.
5. yt-dlp is designed to be never interactive unless specifically requested by `-f-` etc

**Comment by bashonly** [comment]
> So the data is meaningful to the user and potentially very valuable, especially if the source material is not available for download any longer and thus can't be obtained anymore.

Besides livestream downloads, I can't think of any temporary files generated by yt-dlp that would be both valuable and irreplaceable. So IMO this becomes a moot issue once #6705 is resolved

**Comment by pukkandan** [comment]
So, you are saying yt-dlp should just error if a temporary file already exists? This would break the ability to resume downloads among other things.

**Comment by dissona** [comment]
yep very annoying behavior, i have lost 20+gb of download due to this and resuming a download on a .part file from a twitch VOD

this is also useless and ignored:
`--no-continue                   Do not resume partially downloaded
                                    fragments. If the file is not fragmented,
                                    restart download of the entire file`

**Comment by dirkf** [comment]
Somewhat related: #6501

**Comment by pukkandan** [comment]
> auto append " (1)" to it

https://github.com/yt-dlp/yt-dlp/issues/5485
